### ISSUE TITLE
Fail to tag ### Headers inside list 

### ISSUE THREAD

**Issue by v3ss0n** [atom/language-gfm#100]
```gfm
## Activities

- List of Events those are public
- ### Filtering

```
![fail-to-tag](https://cloud.githubusercontent.com/assets/419606/7658861/dc2c1e34-fb63-11e4-9941-10376a154715.png)

**Comment by v3ss0n** [comment]
same goes for all other markdown syntaxes , they are not highlighted inside lists

**Comment by lee-dohm** [comment]
I don't believe this is a bug. If you look at the original [Markdown specification for headers](http://daringfireball.net/projects/markdown/syntax#header), it specifically states that ATX-style headers, i.e. the ones that start with hash marks, "use 1-6 hash characters *at the start of the line*" (emphasis added).

Also, for example:

* ### Filtering

:point_up: isn't a header. Whereas:

* foo bar baz

    ### Filtering

* foo bar baz

:point_up: is a header that is contained within a list.

This is the issue with Markdown and the reason why the [CommonMark project](http://commonmark.org/) was started. Markdown is not rigorously specified, so edge cases like these are up for interpretation.

**Comment by v3ss0n** [comment]
Ah i c , that was not supported in old markdown ? But the renderer , renders it so i thought it was a bug in syntax highlight.

**Comment by lee-dohm** [comment]
The original Markdown is awesome for what it was originally designed for, taking simple conventions patterned after email conversations and turning them into simple HTML documents. It wasn't designed for all the various complex documents that people are trying to use it for ten years later :smile:

And Atom's Markdown flavor is GitHub-flavored Markdown. You can find various references for it in the GitHub Help documentation like [Mastering Markdown](https://guides.github.com/features/mastering-markdown/). If you stick to the formatting that the examples directly show, then things will render as expected. If you wander off into the edge cases, then the various Markdown parsers start to disagree.

For some background, see Jeff Atwood's [blog post announcing Standard-Flavored Markdown, renamed later to CommonMark](http://blog.codinghorror.com/standard-flavored-markdown/). For example:

> Just for that, I count *fifteen* different rendered outputs from 22 different Markdown parsers.
